File: scripts/generate_assets.js
/**
 * Kinetik - Placeholder Asset Generator
 * Run: node scripts/generate_assets.js
 * 
 * Generates minimal valid PNG files for Expo app assets
 */
const fs = require('fs');
const path = require('path');
const zlib = require('zlib');

function createPNG(width, height, r, g, b) {
  // PNG Signature
  const signature = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);
  
  // IHDR chunk
  const ihdrData = Buffer.alloc(13);
  ihdrData.writeUInt32BE(width, 0);
  ihdrData.writeUInt32BE(height, 4);
  ihdrData[8] = 8;  // bit depth
  ihdrData[9] = 2;  // color type (RGB)
  ihdrData[10] = 0; // compression
  ihdrData[11] = 0; // filter
  ihdrData[12] = 0; // interlace
  
  const ihdrType = Buffer.from('IHDR');
  const ihdrCrcData = Buffer.concat([ihdrType, ihdrData]);
  const ihdrCrc = crc32(ihdrCrcData);
  
  const ihdrLength = Buffer.alloc(4);
  ihdrLength.writeUInt32BE(ihdrData.length);
  const ihdrCrcBuf = Buffer.alloc(4);
  ihdrCrcBuf.writeUInt32BE(ihdrCrc);
  
  // IDAT chunk - pixel data
  const rawData = [];
  for (let y = 0; y < height; y++) {
    rawData.push(0); // filter byte (none)
    for (let x = 0; x < width; x++) {
      rawData.push(r, g, b);
    }
  }
  
  const compressed = zlib.deflateSync(Buffer.from(rawData));
  const idatType = Buffer.from('IDAT');
  const idatCrcData = Buffer.concat([idatType, compressed]);
  const idatCrc = crc32(idatCrcData);
  
  const idatLength = Buffer.alloc(4);
  idatLength.writeUInt32BE(compressed.length);
  const idatCrcBuf = Buffer.alloc(4);
  idatCrcBuf.writeUInt32BE(idatCrc);
  
  // IEND chunk
  const iendType = Buffer.from('IEND');
  const iendCrc = crc32(iendType);
  const iendLength = Buffer.alloc(4);
  iendLength.writeUInt32BE(0);
  const iendCrcBuf = Buffer.alloc(4);
  iendCrcBuf.writeUInt32BE(iendCrc);
  
  return Buffer.concat([
    signature,
    ihdrLength, ihdrType, ihdrData, ihdrCrcBuf,
    idatLength, idatType, compressed, idatCrcBuf,
    iendLength, iendType, iendCrcBuf,
  ]);
}

function crc32(data) {
  let crc = 0xFFFFFFFF;
  for (let i = 0; i < data.length; i++) {
    crc ^= data[i];
    for (let j = 0; j < 8; j++) {
      crc = (crc >>> 1) ^ (crc & 1 ? 0xEDB88320 : 0);
    }
  }
  return (crc ^ 0xFFFFFFFF) >>> 0;
}

// Generate assets
const assetsDir = path.join(__dirname, '..', 'packages', 'mobile', 'src', 'assets', 'images');
fs.mkdirSync(assetsDir, { recursive: true });

console.log('Generating placeholder assets...');

// icon.png - 1024x1024 dark background
fs.writeFileSync(path.join(assetsDir, 'icon.png'), createPNG(1024, 1024, 10, 10, 15));
console.log('  ✓ icon.png (1024x1024)');

// splash.png - 1242x2436 dark background
fs.writeFileSync(path.join(assetsDir, 'splash.png'), createPNG(1242, 2436, 10, 10, 15));
console.log('  ✓ splash.png (1242x2436)');

// adaptive-icon.png - 1024x1024
fs.writeFileSync(path.join(assetsDir, 'adaptive-icon.png'), createPNG(1024, 1024, 30, 30, 40));
console.log('  ✓ adaptive-icon.png (1024x1024)');

console.log('\n✅ All placeholder assets generated!');
console.log('   Run "npm install" in the project root to install dependencies.');

File: scripts/generate_assets.py
import struct
import zlib
import os
import sys

def create_png(width, height, r, g, b, filepath):
    """Create a minimal valid PNG with a solid color."""
    # PNG signature
    signature = b'\x89PNG\r\n\x1a\n'
    
    # IHDR chunk
    ihdr_data = struct.pack('>IIBBBBB', width, height, 8, 2, 0, 0, 0)  # 8-bit RGB
    ihdr_crc = zlib.crc32(b'IHDR' + ihdr_data) & 0xFFFFFFFF
    ihdr_chunk = struct.pack('>I', len(ihdr_data)) + b'IHDR' + ihdr_data + struct.pack('>I', ihdr_crc)
    
    # IDAT chunk - raw pixel data
    raw_data = b''
    for y in range(height):
        raw_data += b'\x00'  # filter byte
        for x in range(width):
            raw_data += struct.pack('BBB', r, g, b)
    
    compressed = zlib.compress(raw_data)
    idat_crc = zlib.crc32(b'IDAT' + compressed) & 0xFFFFFFFF
    idat_chunk = struct.pack('>I', len(compressed)) + b'IDAT' + compressed + struct.pack('>I', idat_crc)
    
    # IEND chunk
    iend_crc = zlib.crc32(b'IEND') & 0xFFFFFFFF
    iend_chunk = struct.pack('>I', 0) + b'IEND' + struct.pack('>I', iend_crc)
    
    with open(filepath, 'wb') as f:
        f.write(signature + ihdr_chunk + idat_chunk + iend_chunk)

if __name__ == '__main__':
    base = sys.argv[1] if len(sys.argv) > 1 else 'D:/react-projects/kinetik/packages/mobile/src/assets/images'
    os.makedirs(base, exist_ok=True)
    
    print(f"Generating assets in: {base}")
    
    # icon.png - 1024x1024 dark bg
    create_png(1024, 1024, 10, 10, 15, os.path.join(base, 'icon.png'))
    print("  Created icon.png (1024x1024)")
    
    # splash.png - 1242x2436 dark bg
    create_png(1242, 2436, 10, 10, 15, os.path.join(base, 'splash.png'))
    print("  Created splash.png (1242x2436)")
    
    # adaptive-icon.png - 1024x1024 slightly lighter
    create_png(1024, 1024, 30, 30, 40, os.path.join(base, 'adaptive-icon.png'))
    print("  Created adaptive-icon.png (1024x1024)")
    
    print("Done! All placeholder assets created successfully.")
